internal/build/job/maker.go

```diff
@@ -129,6 +129,11 @@ func (m *maker) specTemplate(
 		kanikoImage += ":" + buildConfig.KanikoParams.Tag
 	}
 
+	nodeSelector := mld.Selector
```

Comment:
why do we need to set the nodeSelector at all for Builds? Why not let kubernetes decide where to run the Build job?

Reply:
This is a good point. Removing it completely will also solve https://github.com/kubernetes-sigs/kernel-module-management/issues/140#issuecomment-1446030028

Reply:
> Why not let kubernetes decide where to run the Build job?

This comes from [that issue](https://github.com/kubernetes-sigs/kernel-module-management/issues/140). You may want to reserve nodes equipped with expensive hardware for some workloads only, for example.

Reply:
I don't see in that issue any mention of nodes with expensive hardware. They just say that it would be nice if build/sign can also run on other nodes, and not only those with specific hardware.  Also,  allowing kuberentes to pick the running node makes use of the schedule, which probably has much more data to decide which nodes are less "overloaded"

Reply:
> Also, allowing kuberentes to pick the running node makes use of the schedule, which probably has much more data to decide which nodes are less "overloaded"

This change makes a user able to do just that, with `selector: {}` in the `build` section.